### PR TITLE
form dynamic conversion

### PR DESCRIPTION
**Pull request by bso-odoo**

### CODE REVIEW

File: addons/website/static/src/snippets/s_website_form/000.js [hunk 1]
```diff
@@ -63,7 +63,14 @@ export class Form extends Interaction {
         },
         ".s_website_form_field[data-type=binary]": {
             "t-on-click": this.clickFileDelete, // delegate on ".o_file_delete"
-        }
+        },
+        ".s_website_form_field": {
+            "t-on-input": (el) => { this.onFieldInputDebounced(el); },
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_website_form/000.js]: you can simplify:
```js
  "t-on-input": this.onFieldInputDebounced,
```

File: addons/website/static/src/snippets/s_website_form/000.js [hunk 2]
```diff
@@ -78,6 +85,7 @@ export class Form extends Interaction {
         // TODO Translation behavior.
         this.editTranslations = false;
         // this.editTranslations = !!this.getContext(true).edit_translations;
+        this.onFieldInputDebounced = debounce(this.onFieldInput.bind(this), 400);
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_website_form/000.js]: note that it is likely unsafe => the interaction can be destroyed in the debounce delay. we need to add a check in the method to protect it

Comment (bso-odoo) [reply addons/website/static/src/snippets/s_website_form/000.js]: The `debounce` has actually become absurd: the handler is empty now, it is there only to trigger the implicit `updateDom` after the event handling.

File: addons/website/static/src/snippets/s_website_form/000.js [hunk 3]
```diff
@@ -63,7 +63,14 @@ export class Form extends Interaction {
...
         ".s_website_form_field[data-type=binary]": {
             "t-on-click": this.clickFileDelete, // delegate on ".o_file_delete"
-        }
+        },
+        ".s_website_form_field": {
+            "t-on-input": this.onFieldInput,
+            "t-att-class": (el) => ({ "d-none": !this.isFieldVisible(el) }),
+        },
+        ".s_website_form_field .s_website_form_input": {
+            "t-att-disabled": (el) => { return !this.isInputVisible(el) || undefined; },
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_website_form/000.js]: for simple functions, you can simplify:
```js
            "t-att-disabled": (el) => !this.isInputVisible(el) || undefined,
```

**Pull request merged**